datafusion/core/tests/parquet/mod.rs

```diff
@@ -740,6 +741,32 @@ fn make_utf8_batch(value: Vec<Option<&str>>) -> RecordBatch {
     .unwrap()
 }
 
+fn make_dictionary_batch(strings: Vec<&str>) -> RecordBatch {
+    let keys = Int32Array::from_iter(0..strings.len() as i32);
+
+    let utf8_values = StringArray::from(strings.clone());
+    let utf8_dict = DictionaryArray::new(keys.clone(), Arc::new(utf8_values));
+
+    let large_utf8 = LargeStringArray::from(strings.clone());
+    let large_utf8_dict = DictionaryArray::new(keys.clone(), Arc::new(large_utf8));
+
+    let binary =
+        BinaryArray::from_iter_values(strings.iter().cloned().map(|v| v.as_bytes()));
+    let binary_dict = DictionaryArray::new(keys.clone(), Arc::new(binary));
+
+    let large_binary =
```

Comment:
I noticed that this PR adds data for `Binary` / `LargeBinary` as well, but not the code / tests

Maybe we could either add that feature in this PR or else file a ticket to track adding it in as a follow on PR